managed/models/database.go

```diff
@@ -1094,55 +1094,42 @@ var databaseSchema = [][]string{
-		`UPDATE agents SET qan_options = jsonb_set(qan_options, '{comments_parsing_disabled}', to_jsonb(comments_parsing_disabled));`,
+		`UPDATE agents SET qan_options['max_query_length'] = to_jsonb(max_query_length)`,
+		`UPDATE agents SET qan_options['max_query_log_size'] = to_jsonb(max_query_log_size)`,
+		`UPDATE agents SET qan_options['query_examples_disabled'] = to_jsonb(query_examples_disabled)`,
+		`UPDATE agents SET qan_options['comments_parsing_disabled'] = to_jsonb(comments_parsing_disabled)`,
 		`ALTER TABLE agents DROP COLUMN max_query_length`,
 		`ALTER TABLE agents DROP COLUMN max_query_log_size`,
 		`ALTER TABLE agents DROP COLUMN query_examples_disabled`,
 		`ALTER TABLE agents DROP COLUMN comments_parsing_disabled`,
 
-		`UPDATE agents SET aws_options = jsonb_set(aws_options, '{aws_access_key}', to_jsonb(aws_access_key));`,
-		`UPDATE agents SET aws_options = jsonb_set(aws_options, '{aws_secret_key}', to_jsonb(aws_secret_key));`,
-		`UPDATE agents SET aws_options = jsonb_set(aws_options, '{rds_basic_metrics_disabled}', to_jsonb(rds_basic_metrics_disabled));`,
-		`UPDATE agents SET aws_options = jsonb_set(aws_options, '{rds_enhanced_metrics_disabled}', to_jsonb(rds_enhanced_metrics_disabled));`,
+		`UPDATE agents SET aws_options['aws_access_key'] = to_jsonb(aws_access_key);`,
+		`UPDATE agents SET aws_options['aws_secret_key'] = to_jsonb(aws_secret_key);`,
+		`UPDATE agents SET aws_options['rds_basic_metrics_disabled'] = to_jsonb(rds_basic_metrics_disabled);`,
+		`UPDATE agents SET aws_options['rds_enhanced_metrics_disabled'] = to_jsonb(rds_enhanced_metrics_disabled);`,
 		`ALTER TABLE agents DROP COLUMN aws_access_key`,
 		`ALTER TABLE agents DROP COLUMN aws_secret_key`,
 		`ALTER TABLE agents DROP COLUMN rds_basic_metrics_disabled`,
 		`ALTER TABLE agents DROP COLUMN rds_enhanced_metrics_disabled`,
 
-		`UPDATE agents SET mysql_options = jsonb_set(mysql_options, '{table_count}', to_jsonb(table_count));`,
-		`UPDATE agents SET mysql_options = jsonb_set(mysql_options, '{table_count_tablestats_group_limit}', to_jsonb(table_count_tablestats_group_limit));`,
+		`UPDATE agents SET mysql_options['table_count'] = to_jsonb(table_count);`,
+		`UPDATE agents SET mysql_options['table_count_tablestats_group_limit'] = to_jsonb(table_count_tablestats_group_limit);`,
 		`ALTER TABLE agents DROP COLUMN table_count`,
 		`ALTER TABLE agents DROP COLUMN table_count_tablestats_group_limit`,
-
-		`UPDATE settings SET settings = jsonb_set(settings, '{encrypted_items}', 
-			'[
-				"pmm-managed.agents.username", 
-				"pmm-managed.agents.password", 
-				"pmm-managed.agents.agent_password", 
-				"pmm-managed.agents.aws_options", 
-				"pmm-managed.agents.azure_options", 
-				"pmm-managed.agents.mongo_options",
-				"pmm-managed.agents.mysql_options", 
-				"pmm-managed.agents.postgresql_options"
-			]'::jsonb
-		)
-		WHERE settings ? 'encrypted_items';`,
```

Comment:
Will this work when someone migrate from v3 beta?

Reply:
we don't support upgrade from v3 beta